extensions/sql-database-projects/src/models/deploy/deployService.ts

```diff
@@ -0,0 +1,389 @@
+				if (validationResult.validated) {
+					this.logToOutput(constants.retrySucceedMessage(name, formatResult(result)));
+					return result;
+				} else {
+					this.logToOutput(constants.retryFailedMessage(name, formatResult(result), validationResult.errorMessage));
+				}
+
+			} catch (err) {
+				this.logToOutput(constants.retryMessage(name, err));
+			}
+		}
+
+		return undefined;
+	}
+
+	private logToOutput(message: string): void {
+		if (this._outputChannel) {
+			this._outputChannel.appendLine(message);
+		}
+	}
+
+	private async createCommands(mssqlFolderPath: string, commandsFolderPath: string, dockerFilePath: string, startFilePath: string, imageLabel: string): Promise<void> {
+		// Create mssql folders if doesn't exist
+		//
+		this.createFolderIfNotExist(mssqlFolderPath);
+		this.createFolderIfNotExist(commandsFolderPath);
+
+		// Start command
+		//
+		await this.createFile(startFilePath, 'echo starting the container!');
+		if (os.platform() !== 'win32') {
+			await this.executeCommand(`chmod +x ${startFilePath}`);
+		}
+
+		// Create the Dockerfile
+		//
+		await this.createFile(dockerFilePath,
+			`
+FROM ${constants.dockerBaseImage}
+ENV ACCEPT_EULA=Y
+ENV MSSQL_PID=Developer
+LABEL ${imageLabel}
+RUN mkdir -p /opt/sqlproject
+COPY ${constants.mssqlFolderName}/${constants.commandsFolderName}/ /opt/commands
+RUN ["/bin/bash", "/opt/commands/start.sh"]
+`);
+	}
+
+	private createFolderIfNotExist(folderName: string): void {
+		if (!fse.existsSync(folderName)) {
+			fse.mkdirSync(folderName);
+		}
+	}
+
+	private async createFile(filePath: string, content: string): Promise<void> {
+		this.logToOutput(`Creating file ${filePath}, content:${content}`);
+		await fse.writeFile(filePath, content);
+	}
+
+	public async cleanDockerObjects(commandToGetObjects: string, commandsToClean: string[]): Promise<void> {
+		let currentIds = await this.executeCommand(commandToGetObjects);
+		if (currentIds) {
+			const ids = currentIds.split(/\r?\n/);
+			for (let index = 0; index < ids.length; index++) {
+				const id = ids[index];
+				if (id) {
+					for (let commandId = 0; commandId < commandsToClean.length; commandId++) {
+						const command = commandsToClean[commandId];
+						await this.executeCommand(`${command} ${id}`);
+
+					}
+				}
+			}
+		}
+	}
+
+	private async executeCommand(cmd: string, timeout: number = 1800000): Promise<string> {
+		return new Promise<string>((resolve, reject) => {
+			this.logToOutput(`    > ${cmd}`);
+			let child = childProcess.exec(cmd, {
+				timeout: timeout
+			}, (err, stdout) => {
+				if (err) {
+					reject(err);
+				} else {
+					resolve(stdout);
+				}
+			});
+
+			// Add listeners to print stdout and stderr if an output channel was provided
+
+			if (child?.stdout) {
+				child.stdout.on('data', data => { this.outputDataChunk(data, '    stdout: '); });
+			}
+			if (child?.stderr) {
+				child.stderr.on('data', data => { this.outputDataChunk(data, '    stderr: '); });
+			}
+		});
+	}
```

Comment:
Not something to do in this PR, but more of a mental note/task: given that we now have at least three instances of this pattern in the database projects extension (local deploy, build, autorest), we should probably genericize this pattern of executing a command and capturing output.